### ISSUE TITLE
Split and trim GPX track at certain track point

### ISSUE THREAD

**Issue by littlehelper** [osmandapp/OsmAnd#6078]
Hi,
I use Osmand quiet often to record my hiking and mountain biking tours as GPX tracks. Sometimes, back at the car I forget to stop the recording and therefore while driving home the track becomes too long and the statistics like travel speed are totally screwed up.
Please add the possibility to split such recorded tracks at a certain track point into two tracks or track files, in a way that one of them can also be deleted easily afterwards.
I could imagine to select the certain track point to split the track in the analyze on map view using the elevation profile cursor and/or in the track edit line view using the track point list.
Thanks.

**Comment by WinterB1860** [comment]
I would like to second this request. I forget to stop recording often enough and if I wish to export the track to OSM I have to spend time deleting what may be a significant number of points, or just trash the track itself. If I could split the track I can delete the remainder I do not wish to upload with one fell swoop

**Comment by vshcherb** [comment]
Merged https://github.com/osmandapp/Osmand/issues/8099

**Comment by Barthwo** [comment]
Also very important for me!

**Comment by littlehelper** [comment]
The GPX track point editor "Edit line" became quite handy for selecting and moving or deleting single track points. Deletion of track segments became also possible... thanks a lot!

To further improve, in situations as decribed in the opening post I would love to have the following additional options in the track points' context menu after selecting one:
- "Split" to split the current track segment into two at the selected track point
- "Delete before" to delete all track points before the selected track point from the current track segment
- "Delete after" to delete all track points after the selected track point from the current track segment

**Comment by vshcherb** [comment]
In new version trim before and trim after is possible

**Comment by notflip** [comment]
Where can I find the edit mode to remove a part of the track? Thank you!

**Comment by vshcherb** [comment]
As of today you can click on a Point on the map and you will see Trim Before / After

**Comment by notflip** [comment]
> As of today you can click on a Point on the map and you will see Trim Before / After

Amazing! This app exceeded all my expectations of a mapping app. Thank you for your work on this.

**Comment by Barthwo** [comment]
Not just on the map but the track has to be "opened" for it.

I really looked for this function for some time until I found it.

Wolfgang

Am 02.10.2020 um 15:26 schrieb vshcherb:

**Comment by scaidermern** [comment]
> hi! I was wondering, could you possibly share the menu interface that allows gpx track editing? I have been searching the gui and the osmand faq on the website - was not able to trim my tracks(

Go to My Places / Tracks, tap on the track. Then tap on the pencil icon, this will enter edit mode. Zoom to the location you want to edit. If you zoom in close enough you will see blue circles. Tap on the blue circle where you want to trim/split the track. A menu will open allowing you to trim and split the track. Here you can also move track points or add additional points.

**Comment by Wonderer0** [comment]
> To further improve, in situations as decribed in the opening post I would love to have the following additional options in the track points' context menu after selecting one:
    "Split" to split the current track segment into two at the selected track point
    "Delete before" to delete all track points before the selected track point from the current track **segment**
    "Delete after" to delete all track points after the selected track point from the current track **segment**

> In new version trim before and trim after is possible

Unfortunately Trim Before and Trim After trim to the beginning/end of the whole track, not just the current segment, even when the "Join segments" option is off.  How do I delete an unwanted segment from between wanted ones?  (e.g. one made exploring a side path but returning to the original place in the main track). (Using 4.4.7 in Android)

**Comment by scaidermern** [comment]
> Unfortunately Trim Before and Trim After trim to the beginning/end of the whole track, not just the current segment, even when the "Join segments" option is off.

Good point. I've encountered the same problem while trying to delete an intermediate segment.

Please make the trim options apply only to the current segment.
(I guess this should go into a new ticket. @Wonderer0 can you create one?)

**Comment by pebogufi** [comment]
> How do I delete an unwanted segment from between wanted ones? (e.g. one made exploring a side path but returning to the original place in the main track). (Using 4.4.7 in Android)

Go to the track view where you can see the segments. The segment option contains a Delete.

**Comment by Wonderer0** [comment]
Thanks @pebogufi, that method seems to be working OK now.  When I had tried that previously on a more complex track with more segments they all seemed to be the same (Distance, Time span etc) and whichever I clicked Delete on it popped up "Delete segment 1?", and nothing happened when I confirmed Yes.

After splitting the track is it possible to display the list of segments without having to first save the track and then reselect it again?  Confusingly, clicking on the track icon (bottom left corner) while in "Plan route" mode displays the list of points, not segments.
